packages/compass-crud/src/components/table-view/cell-renderer.tsx

```diff
@@ -265,121 +200,225 @@ class CellRenderer
+  } else if (element.isAdded()) {
+    cellState = ADDED;
+  } else if (element.isModified()) {
+    cellState = EDITED;
+  } else {
+    cellState = VALID;
   }
 
-  render() {
-    let element;
-    let className = BEM_BASE;
-    let canUndo = false;
-    let canExpand = false;
-
-    if (!this.editable) {
-      element = '';
-      className = `${className}-${UNEDITABLE}`;
-    } else if (this.isEmpty || this.isDeleted) {
-      element = 'No field';
-      className = `${className}-${EMPTY}`;
-    } else if (!this.element.isCurrentTypeValid()) {
-      element = this.renderInvalidCell();
-      className = `${className}-${INVALID}`;
-      canUndo = true;
-    } else if (this.element.isRemoved()) {
-      element = 'Deleted field';
-      className = `${className}-${DELETED}`;
-      canUndo = true;
-    } else {
-      element = this.renderValidCell();
-      if (this.element.isAdded()) {
-        className = `${className}-${ADDED}`;
-        canUndo = true;
-      } else if (this.element.isModified()) {
-        className = `${className}-${EDITED}`;
-        canUndo = true;
+  const handleUndo = useCallback(
+    (event: React.MouseEvent) => {
+      event.stopPropagation();
+      if (!element) {
+        return;
+      }
+      const oid: string = node.data.hadronDocument.getStringId();
+      if (cellState === ADDED) {
+        setIsDeleted(true);
+        const isArray =
+          !element.parent?.isRoot() && element.parent?.currentType === 'Array';
+        elementRemoved(String(element.currentKey), oid, isArray);
+      } else if (cellState === DELETED) {
+        elementAdded(String(element.currentKey), element.currentType, oid);
+      } else {
+        elementTypeChanged(String(element.currentKey), element.type, oid);
       }
-      canExpand =
-        this.element.currentType === 'Object' ||
-        this.element.currentType === 'Array';
+      element.revert();
+    },
+    [
+      element,
+      node.data.hadronDocument,
+      elementRemoved,
+      elementAdded,
+      elementTypeChanged,
+    ]
+  );
+
+  const handleDrillDown = useCallback(
+    (event: React.MouseEvent) => {
+      event.stopPropagation();
+      if (!element) {
+        return;
+      }
+      drillDown(node.data.hadronDocument as Document, element);
+    },
+    [drillDown, node.data.hadronDocument, element]
+  );
+
+  const handleClicked = useCallback(() => {
+    if (node.data.state === 'editing') {
+      api.startEditingCell({
+        rowIndex: node.rowIndex,
+        colKey: column.getColId(),
+      });
     }
+  }, [node, api, column]);
 
-    return (
-      // `ag-grid` renders this component outside of the context chain
-      // so we re-supply the dark mode theme here.
-      <LeafyGreenProvider darkMode={this.props.darkMode}>
+  return (
+    // `ag-grid` renders this component outside of the context chain
+    // so we re-supply the dark mode theme here.
+    <LeafyGreenProvider darkMode={darkMode}>
+      <div>
```

Comment:
It's maybe a diff making it harder to understand for me, but is there any specific reason we added a div here?